File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> temp;
        
        sort(nums1.begin(),nums1.end());
        sort(nums2.begin(),nums2.end());
        int n = nums1.size();
        int m = nums2.size();
        int i=0,j=0;
        while(i<n && j<m){
            if(nums1[i]<nums2[j]){
                i++;
            }
            else if(nums2[j]<nums1[i]){
                j++;
            }
            else {
                temp.insert(nums1[i]);
                i++;
                j++;
            }
        }
        vector<int> ans;
        for(auto t : temp)
            ans.push_back(t);
        return ans;
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        Set<Integer> ans = new HashSet<>();
        int i=0,j=0;
        int len1 = nums1.length;
        int len2 = nums2.length;
        
        while(i<len1 && j<len2){
            if(nums1[i]==nums2[j]){
                ans.add(nums1[i]);
                i++;
                j++;
            }
            else if(nums1[i]<nums2[j])
                i++;
            else j++;
        }
        int[] a = new int[ans.size()];
        i=0;
        for(Integer aa : ans){
            a[i++]=aa;
        }
        return a;
    }
}